pubsub/server.go

```diff
@@ -19,33 +19,33 @@ import (
 
 type ServerConfig struct {
 	// Size of the ws read buffer
-	readBufferSize int
+	ReadBufferSize int
 	// Size of the ws write buffer
-	writeBufferSize int
+	WriteBufferSize int
 	// Maximum number of pending messages to send to a peer.
-	maxPendingMessages int
+	MaxPendingMessages int
 	// Maximum message size in bytes allowed from peer.
-	maxMessageSize int64
+	MaxMessageSize int64
 	// Time allowed to write a message to the peer.
-	writeWait time.Duration
+	WriteWait time.Duration
 	// Time allowed to read the next pong message from the peer.
-	pongWait time.Duration
+	PongWait time.Duration
 	// Send pings to peer with this period. Must be less than pongWait.
-	pingPeriod time.Duration
+	PingPeriod time.Duration
 	// ReadHeaderTimeout is the maximum duration for reading a request.
-	readHeaderTimeout time.Duration
+	ReadHeaderTimeout time.Duration
 }
 
-func NewDefaultServerConfig() ServerConfig {
-	return ServerConfig{
-		readBufferSize:     readBufferSize,
-		writeBufferSize:    writeBufferSize,
-		maxPendingMessages: maxPendingMessages,
-		maxMessageSize:     maxMessageSize,
-		writeWait:          writeWait,
-		pongWait:           pongWait,
-		pingPeriod:         (9 * pongWait) / 10,
-		readHeaderTimeout:  readHeaderTimeout,
+func NewDefaultServerConfig() *ServerConfig {
+	return &ServerConfig{
+		ReadBufferSize:     ReadBufferSize,
+		WriteBufferSize:    WriteBufferSize,
+		MaxPendingMessages: MaxPendingMessages,
+		MaxMessageSize:     MaxMessageSize,
+		WriteWait:          WriteWait,
+		PongWait:           PongWait,
+		PingPeriod:         (9 * PongWait) / 10,
+		ReadHeaderTimeout:  ReadHeaderTimeout,
 	}
 }
 
```

Comment:
I think I could argue that `CheckOrigin` should eventually be in the config but it is good for now.